src/MongoDB.Driver.Core/Core/Operations/FindOneAndUpdateOperation.cs

```diff
@@ -173,6 +186,13 @@ internal override BsonDocument CreateCommand(ICoreSessionHandle session, Connect
         {
             var serverVersion = connectionDescription.ServerVersion;
             Feature.Collation.ThrowIfNotSupported(serverVersion, Collation);
+            if (Feature.HintForFindAndModifyFeature.ShouldThrowIfNeeded(serverVersion))
```

Comment:
I think we shouldn't keep complex logic in feature classes and collation case is different, because it has no differentiation between supported version and version range which must throw exception.

Reply:
I think we shouldn't keep complex logic in feature classes and collation case is different, because it has no differentiation between supported version and version range which must throw exception.